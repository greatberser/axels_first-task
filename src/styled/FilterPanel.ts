import styled from 'styled-components';
import { TextField } from '@mui/material';

export const FilterRow = styled.div`
  display: flex;
  gap: 16px;
  justify-content: center;
  margin-bottom: 16px;
`;

export const SelectInput = styled(TextField)`
  width: 200px;
`;
