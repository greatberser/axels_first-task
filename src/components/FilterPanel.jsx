import { Box, Button, TextField } from '@mui/material';
import styled from 'styled-components';

const FilterRow = styled(Box)`
  display: flex;
  gap: 16px;
  justify-content: center;
  margin-bottom: 16px;
`;

const FilterInput = styled(TextField)`
  width: 200px;
`;

const categories = [
  'All',
  'Food',
  'Transport',
  'Entertainment',
  'Utilities',
  'Health',
];

const FilterPanel = () => {
  return (
    <FilterRow>
      <TextField type="date" variant="outlined" size="small" />
      <FilterInput select label="Category" variant="outlined" size="small">
        {categories.map((category) => (
          <option key={category} value={category}>
            {category}
          </option>
        ))}
      </FilterInput>
      <Button variant="contained" color="primary">
        Apply Filters
      </Button>
    </FilterRow>
  );
};

export default FilterPanel;
