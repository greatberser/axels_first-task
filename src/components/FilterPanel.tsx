import { useState } from 'react';
import { Button, MenuItem, TextField } from '@mui/material';

import { FilterRow, SelectInput } from '../styled/FilterPanel';

export type FilterData = {
  date: string;
  category: string;
};

type FilterPanelProps = {
  onFilter: (data: FilterData) => void;
};

const categories = [
  'All',
  'Food',
  'Transport',
  'Entertainment',
  'Utilities',
  'Health',
];

const FilterPanel: React.FC<FilterPanelProps> = ({ onFilter }) => {
  const [date, setDate] = useState('');
  const [category, setCategory] = useState('All');

  const handleFilter = () => {
    onFilter({
      date,
      category,
    });
  };

  return (
    <FilterRow>
      <TextField
        type="date"
        label="Date"
        InputLabelProps={{ shrink: true }}
        id="date"
        variant="outlined"
        size="small"
        value={date}
        onChange={(e) => setDate(e.target.value)}
      />
      <SelectInput
        select
        label="Category"
        id="category"
        variant="outlined"
        size="small"
        value={category}
        onChange={(e) => setCategory(e.target.value)}
      >
        {categories.map((category) => (
          <MenuItem key={category} value={category}>
            {category}
          </MenuItem>
        ))}
      </SelectInput>
      <Button variant="contained" color="primary" onClick={handleFilter}>
        Apply Filters
      </Button>
    </FilterRow>
  );
};

export default FilterPanel;
