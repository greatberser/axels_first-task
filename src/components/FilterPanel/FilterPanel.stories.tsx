import FilterPanel from './FilterPanel';
import { Meta, StoryObj } from '@storybook/react';

const meta: Meta<typeof FilterPanel> = {
  title: 'Components/FilterPanel',
  component: FilterPanel,
  tags: ['autodocs'],
};

export default meta;

type FilterPanelProps = {
  onFilter: (data: { date: string; category: string }) => void;
};

export const Default: StoryObj<FilterPanelProps> = {
  args: {
    onFilter: (data) => console.log('Filter applied:', data),
  },
};
