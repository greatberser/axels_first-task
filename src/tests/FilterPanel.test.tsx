import { render, screen } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import FilterPanel from '../components/FilterPanel';

describe('FilterPanel', () => {
  it('should match a snapshot', () => {
    const { asFragment } = render(<FilterPanel onFilter={jest.fn()} />);
    expect(asFragment()).toMatchSnapshot();
  });

  it('should call onFilter with correct data', async () => {
    const mockOnFilter = jest.fn();
    render(<FilterPanel onFilter={mockOnFilter} />);
    const user = userEvent.setup();

    const dateInput = screen.getByLabelText(/date/i);
    await user.clear(dateInput);
    await user.type(dateInput, '2023-10-01');

    const categorySelect = screen.getByLabelText(/category/i);
    await user.click(categorySelect);

    const foodOption = await screen.findByText('Food');
    await user.click(foodOption);

    const applyButton = screen.getByRole('button', { name: /apply filters/i });
    await user.click(applyButton);

    expect(mockOnFilter).toHaveBeenCalledWith({
      date: '2023-10-01',
      category: 'Food',
    });
  });
});
