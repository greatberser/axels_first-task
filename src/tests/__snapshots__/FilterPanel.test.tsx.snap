// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`FilterPanel should match a snapshot 1`] = `
<DocumentFragment>
  <div
    class="sc-bRKDuR gwymt"
  >
    <div
      class="MuiFormControl-root MuiTextField-root css-1xp5r68-MuiFormControl-root-MuiTextField-root"
    >
      <label
        class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-sizeSmall MuiInputLabel-outlined MuiFormLabel-colorPrimary MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-sizeSmall MuiInputLabel-outlined css-hvjq6j-MuiFormLabel-root-MuiInputLabel-root"
        data-shrink="true"
        for="date"
        id="date-label"
      >
        Date
      </label>
      <div
        class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-formControl MuiInputBase-sizeSmall css-quhxjy-MuiInputBase-root-MuiOutlinedInput-root"
      >
        <input
          aria-invalid="false"
          class="MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputSizeSmall css-1pzfmz2-MuiInputBase-input-MuiOutlinedInput-input"
          id="date"
          type="date"
          value=""
        />
        <fieldset
          aria-hidden="true"
          class="MuiOutlinedInput-notchedOutline css-18p5xg2-MuiNotchedOutlined-root-MuiOutlinedInput-notchedOutline"
        >
          <legend
            class="css-ex8a5f-MuiNotchedOutlined-root"
          >
            <span>
              Date
            </span>
          </legend>
        </fieldset>
      </div>
    </div>
    <div
      class="MuiFormControl-root MuiTextField-root sc-hvigdm leArfF css-1xp5r68-MuiFormControl-root-MuiTextField-root"
    >
      <label
        class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-sizeSmall MuiInputLabel-outlined MuiFormLabel-colorPrimary MuiFormLabel-filled MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-sizeSmall MuiInputLabel-outlined css-hvjq6j-MuiFormLabel-root-MuiInputLabel-root"
        data-shrink="true"
        for="category"
        id="category-label"
      >
        Category
      </label>
      <div
        class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-formControl MuiInputBase-sizeSmall MuiSelect-root css-quhxjy-MuiInputBase-root-MuiOutlinedInput-root"
      >
        <div
          aria-expanded="false"
          aria-haspopup="listbox"
          aria-labelledby="category-label category"
          class="MuiSelect-select MuiSelect-outlined MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputSizeSmall css-trm6af-MuiNativeSelect-root-MuiSelect-select-MuiInputBase-input-MuiOutlinedInput-input"
          id="category"
          role="combobox"
          tabindex="0"
        >
          All
        </div>
        <input
          aria-hidden="true"
          aria-invalid="false"
          class="MuiSelect-nativeInput css-j0riat-MuiSelect-nativeInput"
          tabindex="-1"
          value="All"
        />
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiSelect-icon MuiSelect-iconOutlined css-1rfqz7c-MuiSvgIcon-root-MuiNativeSelect-root-MuiSelect-icon"
          data-testid="ArrowDropDownIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M7 10l5 5 5-5z"
          />
        </svg>
        <fieldset
          aria-hidden="true"
          class="MuiOutlinedInput-notchedOutline css-18p5xg2-MuiNotchedOutlined-root-MuiOutlinedInput-notchedOutline"
        >
          <legend
            class="css-ex8a5f-MuiNotchedOutlined-root"
          >
            <span>
              Category
            </span>
          </legend>
        </fieldset>
      </div>
    </div>
    <button
      class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-colorPrimary MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-colorPrimary css-74d805-MuiButtonBase-root-MuiButton-root"
      tabindex="0"
      type="button"
    >
      Apply Filters
    </button>
  </div>
</DocumentFragment>
`;
